src/model/event.rs

```diff
@@ -266,27 +267,43 @@ impl<'de> Deserialize<'de> for GuildMembersChunkEvent {
     }
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Serialize)]
 #[non_exhaustive]
 pub struct GuildRoleCreateEvent {
-    #[serde(deserialize_with = "roles::deserialize_event")]
     pub role: Role,
 }
 
+impl<'de> Deserialize<'de> for GuildRoleCreateEvent {
+    /// Custom deserializer to pass entire payload to roles::deserialize_event, does not work using `deserialize_with`
```

Comment:
Why does this no longer work?

Reply:
As `roles::deserialize_event` needs the payload for the entire event, instead of just the `role` field as would be passed using `deserialize_with`

Reply:
Ah, you're changing it back to the original code before you broke it with #1938.